### PR TITLE
[CALCITE-5851] Add LEVENSHTEIN function (enabled in Hive and Spark library)

### PR DESCRIPTION
**Pull request by herunkang2018**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java [hunk 1]
```diff
@@ -688,6 +692,11 @@ public static String reverse(String s) {
     return buf.reverse().toString();
   }
 
+  /** SQL "LEVENSHTEIN(string1, string2)" function. */
```
Comment (NobiGo) [inline core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java:695]: In order to consistent programming style, Please remove double quotation marks.

Comment (herunkang2018) [reply core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java:695]: Thanks for the review, I will fix it soon.

Comment (herunkang2018) [reply core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java]: Fixed.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 2]
```diff
@@ -4320,6 +4320,23 @@ static void checkRlikeFails(SqlOperatorFixture f) {
...
+    f0.checkFails("^levenshtein('abc', 'abc')^",
+        "No match found for function signature LEVENSHTEIN\\(<CHARACTER>, <CHARACTER>\\)",
+        false);
+    final Consumer<SqlOperatorFixture> consumer = f -> {
+      f.checkScalar("levenshtein('', '')", 0, "INTEGER NOT NULL");
+      f.checkScalar("levenshtein('abc', 'abc')", 0, "INTEGER NOT NULL");
+      f.checkScalar("levenshtein('kitten', 'sitting')", 3, "INTEGER NOT NULL");
+      f.checkScalar("levenshtein('frog', 'fog')", 1, "INTEGER NOT NULL");
+      f.checkScalar("levenshtein(_UTF8'\u4F60\u597D', _UTF8'\u4F60\u5F88\u597D')",
+          1, "INTEGER NOT NULL");
```
Comment (tanclary) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:4334]: What is the expected behavior if the second argument is null?

**Review by tanclary — COMMENTED** [review]
Left one question, otherwise I think it looks good

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 2 above)
Comment (herunkang2018) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:4334]: It will return null, I will add a test case for it.

**Comment by herunkang2018** [comment]
@tanclary @NobiGo Thanks a lot for reviewing. All comments resolved, please help to review if you have time.

**Review by NobiGo — COMMENTED** [review]
LGTM, Please rebase the commits. If the CI pass, I will merge it.

**Comment by herunkang2018** [comment]
@NobiGo Thanks for your review. I have rebased yet.

**Pull request merged**